primitives/runtime/src/lib.rs

```diff
@@ -476,16 +496,7 @@ pub enum DispatchError {
 	/// A bad origin.
 	BadOrigin,
 	/// A custom error in a module.
-	Module {
-		/// Module index, matching the metadata module index.
-		index: u8,
-		/// Module specific error value.
-		error: u8,
-		/// Optional error message.
-		#[codec(skip)]
-		#[cfg_attr(feature = "std", serde(skip_deserializing))]
-		message: Option<&'static str>,
-	},
+	Module(ModuleError),
```

Comment:
The only problem I see is that this changes the serde serialization, but I'm not sure we expose this error anywhere over rpc or similar.

Reply:
Yeah, I am not sure if [`#[serde(flatten)]`](https://serde.rs/field-attrs.html#flatten) is supported on the variant. But if it is, that could help here.